cmake/modules/extensions.cmake

```diff
@@ -4999,31 +5023,22 @@ function(add_llext_target target_name)
     # No conversion required, simply copy the object file
     add_custom_command(
       OUTPUT ${output_file}
-      COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_OBJECTS:${target_name}_lib> ${output_file}
-      DEPENDS ${target_name}_lib $<TARGET_OBJECTS:${target_name}_lib>
+      COMMAND ${CMAKE_COMMAND} -E copy ${linked_file} ${output_file}
```

Comment:
This is OT - this PR is not changing current functionality or addressing this topic. 
(My solution would be to move to dynamic libs everywhere, but first need to merge the link code).

Reply:
It could work with multiple files, but it would require wrapping an ld merge command...

```sh
ld -relocatable in0.o in1.o -o out.o
```

There's no add_library command for a statically relocatable elf today, we have static, dynamic, and object as options. CMake makes this somewhat annoying to deal with actually.